### PR TITLE
Use JDK 11 for Helix Project

### PR DESCRIPTION
**Pull request by desaikomal**
### Issues

- [x] My PR addresses the following Helix issues and references them in the PR description:

### Description
Helix has been using JDK 1.8. We want to move to latest JDK but prior to that, we can use JDK11 as intermediate step. This will help us remain current and use all the language optimizations for performance improvements.

We were using Jetty - 9.4.48.v20220622 and Jersey version 2.15.
Both these versions were not compatible with JDK11.
If you go with the latest version of Jetty and latest of Jersey, the REST Server never starts.
So we need to do 2 levels of compatibility:
- with JDK 11 compatibility
- compatibility amongst both of them. 

It took me while to figure out Jetty/Jersey compatibility. In addition some APIs had changed.

### Tests
- The following is the result of the "mvn test" command on the appropriate module:
[ERROR] Failures:
[ERROR]   TestNoThrottleDisabledPartitions.testDisablingTopStateReplicaByDisablingInstance:98 expected:<false> but was:<true>
[ERROR]   TestNoThrottleDisabledPartitions.testNoThrottleOnDisabledInstance:231->setupEnvironment:317->setupCluster:436 » ZkClient
[ERROR]   TestP2PNoDuplicatedMessage.testP2PStateTransitionEnabled:180 expected:<true> but was:<false>
[ERROR]   TestMultiZkConnectionConfig>MultiZkTestBase.afterClass:137 expected:<true> but was:<false>
[ERROR]   TestMultiZkConnectionConfig.testZKHelixManager:295->Object.wait:328->Object.wait:-2 » ThreadTimeout
[ERROR]   TestWagedRebalanceFaultZone.testAddZone:270->validate:318->validateZoneAndTagIsolation:342 expected:<3> but was:<2>
[ERROR]   TestTaskStateModelFactory.testZkClientCreationMultiZk:80 » Bind Address alread...
[INFO]
[ERROR] Tests run: 1325, Failures: 7, Errors: 0, Skipped: 6

I ran all the 7 failures as independent test-runs. Some tests passed on running while some failed. I took a fresh code branch with no changes and ran the same failed tests against it and they failed there as well. So between re-run and clean branch, there are NO NEW failures.

PR-CI run is completely GREEN (including all other components)
https://github.com/apache/helix/actions/runs/4289361614

### Changes that Break Backward Compatibility (Optional)

- My PR contains changes that break backward compatibility or previous assumptions for certain methods or API. They include:

(Consider including all behavior changes for public methods or API. Also include these changes in merge description so that other developers are aware of these changes. This allows them to make relevant code changes in feature branches accounting for the new method/API behavior.)

### Documentation (Optional)

- In case of new functionality, my PR adds documentation in the following wiki page:

(Link the GitHub wiki you added)

### Commits

- My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

File: .github/workflows/Helix-CI.yml [hunk 1]
```diff
@@ -12,14 +12,14 @@ jobs:
...
     steps:
     - uses: actions/checkout@v2
-    - name: Set up JDK 1.8
+    - name: Set up JDK 11
       uses: actions/setup-java@v1
       with:
-        java-version: 1.8
+        java-version: 11
     - name: Build with Maven
-      run: mvn clean install -Dmaven.test.skip.exec=true -DretryFailedDeploymentCount=5
```
Comment (rahulrane50) [inline .github/workflows/Helix-CI.yml:20]: May I ask any reason for removing this "re-run failed tests" flag?

**Comment by rahulrane50** [comment]
Not an expert here but just curious don't we need fe other changes like adding new dependencies in jdk 11, GC options and few newer APIs changes. (reference : https://gist.github.com/dineshbhagat/97eff450e9edbd7d006deb35f28e3fd4)

File: .github/workflows/Helix-CI.yml (hunk 1 above)
Comment (desaikomal) [reply .github/workflows/Helix-CI.yml:20]: i accidently had removed the modification. re-introduced with latest change

**Comment by desaikomal** [comment]
JDK11 - optimization work can start once we have the base line.

File: .github/workflows/Helix-PR-CI.yml [hunk 2]
```diff
@@ -14,14 +14,14 @@ jobs:
...
       uses: actions/setup-java@v1
       with:
-        java-version: 1.8
+        java-version: 11
     - name: Build with Maven
-      run: mvn clean install -Dmaven.test.skip.exec=true -DretryFailedDeploymentCount=5
+      run: mvn clean install -Dmaven.test.skip.exec=true
     - name: Run All Tests
-      run: mvn -q -fae -Dsurefire.rerunFailingTestsCount=5 test
+      run: mvn -q -fae test
```
[the remark this answers is not in the document]
Comment (desaikomal) [reply .github/workflows/Helix-PR-CI.yml]: No, it wasn't. I fixed it. thanks for pointing out.

File: helix-rest/src/main/java/org/apache/helix/rest/server/HelixRestServer.java [hunk 3]
```diff
@@ -283,7 +283,8 @@ private void cleanupResourceConfigs() {
     }
   }
 
-  public void setupSslServer(int port, SslContextFactory sslContextFactory) {
+
+  public void setupSslServer(int port, SslContextFactory.Server sslContextFactory) {
```
Comment (rahulrane50) [inline helix-rest/src/main/java/org/apache/helix/rest/server/HelixRestServer.java:287]: Neat : If Now parent factory class is deprecated and directly server and client class are encouraged to access then should we rename this variable as well to "sslContextServer"? Also may be update documentation/comments here?

**Review by rahulrane50 — APPROVED** [review]
Thanks for starting this @desaikomal overall LGTM!.

**Review by qqu0127 — APPROVED** [review]
LGTM, thanks

**Review by junkaixue — APPROVED** [review]
LGTM! Thanks for working on this!

**Review by mgao0 — APPROVED** [review]
LGTM, could you please add some explanation in the PR description about the jetty and jersey problem, just for future reference.

**Comment by desaikomal** [comment]
Thank you all (@junkaixue , @mgao0 , @rahulrane50 , @qqu0127 ) for the review. 
Change has been approved by @junkaixue , @mgao0 , @rahulrane50 and @qqu0127.  Commit message : Upgrading Helix to use JDK 11.

**Pull request merged**